go_node_engine/requests/ClusterManagerRequests.go

```diff
@@ -23,7 +24,19 @@ func ClusterHandshake(address string, port int) HandshakeAnswer {
 		logger.ErrorLogger().Fatalf("Handshake failed, json encoding problem, %v", err)
 	}
 	jsonbody := bytes.NewBuffer(data)
-	resp, err := http.Post(fmt.Sprintf("http://%s:%d/api/node/register", address, port), "application/json", jsonbody)
+
+	cfg, err := config.GetConfFileManager().Get()
+	if err != nil {
+		logger.ErrorLogger().Fatalf("Could not get config")
+	}
+
+	var resp *http.Response
+	if cfg.ClusterSSL {
+		resp, err = http.Post(fmt.Sprintf("https://%s:%d/api/node/register", address, port), "application/json", jsonbody)
+	} else {
+		resp, err = http.Post(fmt.Sprintf("http://%s:%d/api/node/register", address, port), "application/json", jsonbody)
+	}
```

Comment:
Instead of repeating the call twice in the code, can you have a `proto` variable like the following example?

```suggestion
	proto := "http"
	if cfg.ClusterSSL {
		proto = "https"
	} 
	resp, err = http.Post(fmt.Sprintf("%s://%s:%d/api/node/register",proto, address, port), "application/json", jsonbody)
```